### PR TITLE
rewrite the docs

### PR DESCRIPTION
**Pull request by nslater**

### CODE REVIEW

File: docs/compatibility.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  - However, referencing relations can be done without joins using Doctrine's
+    lazy loading mechanism with subsequent SELECTs (except many-to-many
+    releations)
+
+Column Definitions
+------------------
+
+Unfortunately, when defining CrateDB `object`_ or `array`_ type columns, you
+must construct the DBAL column definition object manually, using the correct
+class names and attributes.
```
[the remark this answers is not in the document]
Comment (chaudum) [reply docs/compatibility.rst]: in general, yes.

File: docs/compatibility.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  - However, referencing relations can be done without joins using Doctrine's
+    lazy loading mechanism with subsequent SELECTs (except many-to-many
+    releations)
+
+Column Definitions
+------------------
+
+Unfortunately, when defining CrateDB `object`_ or `array`_ type columns, you
+must construct the DBAL column definition object manually, using the correct
```
Comment (chaudum) [inline docs/compatibility.rst]: manually -> programmatically

File: docs/compatibility.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+- Joins
+
+  - However, referencing relations can be done without joins using Doctrine's
+    lazy loading mechanism with subsequent SELECTs (except many-to-many
+    releations)
+
+Column Definitions
+------------------
+
+Unfortunately, when defining CrateDB `object`_ or `array`_ type columns, you
```
Comment (chaudum) [inline docs/compatibility.rst]: Primitive column types, such as string, integer, ... can be defined in the regular Doctrine way.

File: docs/compatibility.rst (hunk 2 above)
Comment (chaudum) [reply docs/compatibility.rst]: ~~object~~

Comment (chaudum) [reply docs/compatibility.rst]: using the custom types and attributed provided by the CrateDB DBAL driver.

**Comment by nslater** [comment]
@chaudum thanks for the review! I have addressed all your comments!

there's one final comment from me at the top. do you think you could take a look at it and let me know what you think? ty ty!

File: docs/appendices/compatibility.rst [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
++                +---------------------------------------+------------+
+|                | Connection to multiple CrateDB nodes. | Supported. |
++                +---------------------------------------+------------+
+|                | Default schema selection.             | Supported. |
++                +---------------------------------------+------------+
+|                | CrateDB SQL features:                 | Supported. |
+|                |                                       |            |
+|                | - Joins                               |            |
+|                | - Fulltext indexes                    |            |
++----------------+---------------------------------------+------------+
```
Comment (nslater) [inline docs/appendices/compatibility.rst]: @chaudum @Taliik can you verify the table above?

**Comment by nslater** [comment]
@chaudum @Taliik I have pushed some fixups. can I get another review please? ty!

File: docs/appendices/data-types.rst [hunk 5]
```diff
@@ -15,23 +15,24 @@ This driver maps CrateDB types to the following PHP types:
...
    "`string`__", "`string`__"
-   "`ip`__", ""
+   "`ip`__", "`string`__"
    "`timestamp`__", "`TimestampType`__"
-   "`geo_point`__", ""
-   "`geo_shape`__", ""
+   "`geo_point`__", "`array`__"
+   "`geo_shape`__", "`object`__"
    "`object`__", "`MapType`__"
    "`array`__", "`ArrayType`__"
```
Comment (nslater) [inline docs/appendices/data-types.rst]: @seut can you verify I've done this right, per your comments on trello?

File: docs/appendices/compatibility.rst (hunk 4 above)
Comment (nslater) [reply docs/appendices/compatibility.rst]: verified and fixed. thanks

**Comment by nslater** [comment]
@seut can I get a final review on this to merge it?

last remaining comment is here:

https://github.com/crate/crate-dbal/pull/48#discussion_r186176720

File: .gitignore [hunk 6]
```diff
@@ -1,5 +1,6 @@
 .idea
 .installed.cfg
+.installed.cfg
```
Comment (chaudum) [inline .gitignore]: Is this a duplicated that slipped in?

File: docs/appendices/data-types.rst [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+   "`short`__", "`integer`__"
+   "`integer`__", "`integer`__",
+   "`long`__", "`string`__"
+   "`float`__", "`float`__"
+   "`double`__", "`float`__"
+   "`string`__", "`string`__"
+   "`ip`__", "`string`__"
+   "`timestamp`__", "`TimestampType`__"
+   "`geo_point`__", "`array`__"
+   "`geo_shape`__", "`object`__"
```
Comment (seut) [inline docs/appendices/data-types.rst]: php type is `array`

File: docs/appendices/data-types.rst [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+   "`object`__", "`MapType`__"
```
Comment (seut) [inline docs/appendices/data-types.rst]: php type is `array`

File: docs/appendices/data-types.rst [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+   "`array`__", "`ArrayType`__"
```
Comment (seut) [inline docs/appendices/data-types.rst]: php type is `array`

File: docs/appendices/data-types.rst [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+   "`boolean`__", "`boolean`__"
+   "`byte`__", "`integer`__"
+   "`short`__", "`integer`__"
+   "`integer`__", "`integer`__",
+   "`long`__", "`string`__"
+   "`float`__", "`float`__"
+   "`double`__", "`float`__"
+   "`string`__", "`string`__"
+   "`ip`__", "`string`__"
+   "`timestamp`__", "`TimestampType`__"
```
Comment (seut) [inline docs/appendices/data-types.rst]: php type is `DateTime`

**Comment by nslater** [comment]
@seut okay pushed a fixup! ty!

**Comment by nslater** [comment]
@seut pushed another fixup thanks!

**Pull request merged**